src/app/actions/exportPdf.test.ts

```diff
@@ -0,0 +1,80 @@
+/**
+ * @jest-environment node
+ */
+import * as fs from 'fs';
+
+import { exportSearchResultPdf } from '@/app/actions/exportPdf';
+import { getServerSession } from '@/lib/actionRequest/session';
+import { convertHtmlToPdf } from '@/lib/file';
+import { SessionWithId } from '@/lib/handler/auth.action';
+import { createManySearchResults } from '@/repositories/searchResult';
+import { createUser } from '@/repositories/user';
+import { searchResultFactory } from '@test/factories/searchResult';
+import { userFactory } from '@test/factories/user';
+
+const mimeTypeIdentifierPDF = 'JVBERi0';
+
+jest.mock('../../lib/actionRequest/session');
+jest.mock('../../lib/file');
+
+describe('exportSearchResultPdf', () => {
+  const serverSessionUtilities = { getServerSession };
+  const fileUtilities = { convertHtmlToPdf };
+  const mockedServerSession = serverSessionUtilities as jest.Mocked<{
+    getServerSession: () => Promise<SessionWithId | null>;
+  }>;
+  const mockedFile = fileUtilities as jest.Mocked<{
+    convertHtmlToPdf: () => Promise<Buffer>;
+  }>;
+
+  describe('given user is authorized', () => {
```

Comment:
It should be:

```suggestion
  describe('given user is authenticated', () => {
```

Authorization would be about checking user roles and permissions. There is no such checks in place now 🙈